components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java

```diff
@@ -355,6 +356,12 @@ public SAMLSSOServiceProviderInfoDTO getServiceProviders() throws IdentityExcept
                 providerDTO.setKeyEncryptionAlgorithmURI(providerDO.getKeyEncryptionAlgorithmUri());
                 providerDTO.setCertAlias(providerDO.getCertAlias());
                 providerDTO.setAttributeConsumingServiceIndex(providerDO.getAttributeConsumingServiceIndex());
+
+                if (providerDO.getAttributeConsumingServiceIndex() != null && !providerDO
+                        .getAttributeConsumingServiceIndex().equals("")) {
```

Comment:
Better to use `StringUtils.isNotBlank()` here.

Reply:
@ashensw Fixed